### PR TITLE
Support more general fenced code block

### PR DESCRIPTION
**Pull request by yous**
I usually see fenced code block with a language specifier followed by \`\`\` and some whitespace. This pull request supports highlighting them.

### CODE REVIEW

**Comment by yous** [comment]
I have no idea why the tests fail. Is there anyone can help me to solve this problem?

Also I noticed that the tests fail when I duplicate the 'fenced code block syntax with a language specifier' test. Then is it a problem with vader.vim?

**Comment by shirosaki** [comment]
Buffer local variables which have included file types would cause failures.
Resetting buffer local variables is normally handled by `autocmd`, but `autocmd` doesn't work in test and tests would run in same buffer.
Clearing these variables before each test is a solution.

```patch
diff --git a/test/syntax.vader b/test/syntax.vader
index 044a7d1..74542c1 100644
--- a/test/syntax.vader
+++ b/test/syntax.vader
@@ -1,3 +1,7 @@
+Before:
+   unlet! b:mkd_known_filetypes
+   unlet! b:mkd_included_filetypes
+
 Given markdown;
 a **b** c

```

**Comment by shirosaki** [comment]
`After` also works.

```patch
diff --git a/test/syntax.vader b/test/syntax.vader
index 044a7d1..da97e60 100644
--- a/test/syntax.vader
+++ b/test/syntax.vader
@@ -1,3 +1,7 @@
+After:
+   unlet! b:mkd_known_filetypes
+   unlet! b:mkd_included_filetypes
+
 Given markdown;
 a **b** c

```

**Comment by yous** [comment]
@shirosaki Thank you for the solutions! I just added a `Before` hook.
